src/ListTabulator/index.ts

```diff
@@ -366,353 +445,450 @@ export default class ListTabulator {
-         *   1.1. |My
-         *     1.1.1. Wonderful
-         *     1.1.2. World
-         *
-         * After:
-         * 1. Hello|My
-         *   1.1. Wonderful
-         *   1.2. World
-         */
-        currentItem.after(item);
-      } else {
-        /**
-         * Not the first item
-         *
-         * Before:
-         * 1. Hello
-         *   1.1. My
-         *   1.2. |Dear
-         *     1.2.1. Wonderful
-         *     1.2.2. World
-         *
-         * After:
-         * 1. Hello
-         *   1.1. My|Dear
-         *   1.2. Wonderful
-         *   1.3. World
-         */
-        targetItem.after(item);
-      }
-    });
+      endingHTML = getContenteditableSlice(currentItemContent, currentNode, offset, 'right', true);
+    }
+
+    const itemChildren = currentItem?.querySelector(
+      `.${DefaultListCssClasses.itemChildren}`
+    );
 
     /**
-     * Remove current item element
+     * Create the new list item
      */
-    currentItem.remove();
+    const itemEl = this.renderer!.renderItem(endingHTML, { checked: false });
 
     /**
-     * Restore the caret position
+     * Move new item after current
      */
-    this.caret.restore();
-  }
+    currentItem?.after(itemEl);
 
+    /**
+     * If current item has children, move them to the new item
+     */
+    if (itemChildren) {
+      itemEl.appendChild(itemChildren);
+    }
+
+    this.focusItem(itemEl);
+  }
 
   /**
-   * Reduce indentation for current item
-   *
-   * @param {KeyboardEvent} event - keydown
-   * @returns {void}
+   * Method that is used for merging current item with previous one
+   * Content of the current item would be appended to the previous item
+   * Current item children would not change nesting level
+   * @param currentItem - current item html element
    */
-  shiftTab(event: KeyboardEvent): void {
+  mergeCurrentItemWithPrevious(currentItem: HTMLElement): void {
+    const previousItem = currentItem.previousElementSibling;
+
+    const parentItem = currentItem.closest(`.${DefaultListCssClasses.item}`);
+
     /**
-     * Prevent editor.js behaviour
+     * Check that current item has any previous siblings to be merged with
      */
-    event.stopPropagation();
+    if (!previousItem && !parentItem) {
+      return;
+    }
 
     /**
-     * Prevent browser tab behaviour
+     * Make sure previousItem is an HTMLElement
      */
-    event.preventDefault();
+    if (previousItem && !isHtmlElement(previousItem)) {
+      return;
+    }
 
     /**
-     * Move item from current list to parent list
+     * Lets compute the item which will be merged with current item text
      */
-    this.unshiftItem();
-  }
+    let targetItem: Element | null;
 
+    /**
+     * If there is a previous item then we get a deepest item in its sublists
+     *
+     * Otherwise we will use the parent item
+     */
+    if (previousItem) {
+      const childrenOfPreviousItem = previousItem.querySelectorAll(`.${DefaultListCssClasses.item}`);
```

Comment:
use util for getting children. You have it.

Reply:
i have util only for getting first level child, here i need to get all children to know deepest one, left a comment